src/Endatix.Extensions.Hosting/EndatixWebAppBuilderExtensions.cs

```diff
@@ -0,0 +1,67 @@
+using Ardalis.GuardClauses;
+using Microsoft.AspNetCore.Builder;
+using Logging = Microsoft.Extensions.Logging;
+using Serilog;
+using Serilog.Extensions.Logging;
+using Endatix.Framework.Hosting;
+using Endatix.Extensions.Hosting;
+using System.Runtime.Versioning;
+using Endatix.Core.Configuration;
+using Microsoft.Extensions.Configuration;
+
+namespace Endatix.Setup;
+
+
+public static class EndatixHostBuilderExtensions
+{
+    public static IEndatixApp UseEndatix(this WebApplicationBuilder builder, Logging.ILogger? logger = null)
+    {
+        Guard.Against.Null(builder);
+
+        if (logger == null)
+        {
+            var serilogLogger = new LoggerConfiguration()

```

Comment:
Something small but I think can help readability - extract the constructing of the logger object in a private method.
I suppose with the time `UseEndatix` will become bigger and extracting such actions can be helpful.